### ISSUE TITLE
Sometimes some saved prompt is missing

### ISSUE THREAD

**Issue by steelywing** [Zetaphor/PromptHub#4]
When restart SD, sometimes some of the saved prompt is missing.

**Comment by Zetaphor** [comment]
Only some of them and not all of them?

Are you clearing cache? The prompt history is saved in your browsers localStorage. If you or an extension is clearing your cache that is going to clear the history. I recommend using the export function if you want to make sure they are saved.

**Comment by steelywing** [comment]
Only some of them, I found the cause is, when the extension load the `.json`, it will make some of the `id` become the same number, are you using the timestamp for the `id`? may be that is Firefox bug? can you test in Firefox?

May be using a big random number for the ID? or using the name as `id`?

**Comment by steelywing** [comment]
I vote to use the name as `id`, usually when we save with the same name, we just want to overwrite it, just ask the user if the user want to overwrite.

**Comment by steelywing** [comment]
I think this line is the issue, https://github.com/Zetaphor/PromptHub/blob/master/javascript/interface.js#L100

if
```
first random = 8;
second random = 7;
and the first process use 1ms, then the 2 IDs will be same
```
or you can run
```
for (let i=0;i<20;i++) {
  console.log(Date.now());
}
```
you will see many timestamp is the same, so if 2 random numbers is same, the ID mostly get same.

**Comment by Zetaphor** [comment]
Good catch. I'm working on an update to make this work with the latest updates to Automatic, I will include this fix there
